### ISSUE TITLE
With multiple tweens, calling stop on any of them from within their onComplete callback causes other tweens to never complete

### ISSUE THREAD

**Issue by lmeikle** [tweenjs/tween.js#337]
Hi

There seems to be an issue regarding calling stop on a tween from inside its onComplete callback

Here is a simple example:
https://jsfiddle.net/mpq3hkhe/3/

I create 2 tweens and start them.
When tween 1 completes, it calls stop on itself. This causes tween 2 to never complete.

(You will notice I have also done an another example which works - simply by putting the call to stop in a timeout.)

I know calling stop on tween 1 is not necessary as it has stopped already, I've just done this to demonstrate the issue quickly. 
A real life scenario would more like having a reset function which does loads of stuff including stopping the tween. This reset function can be called mid way through the tween or when the tween completes. Depending on when it is called, it should never have adverse effects on other tweens.

I would like to know whether you think I am wrong in my thinking above, and we must always be very careful not to call stop from the onComplete callback.
Or you agree that there seems to be an issue here, if so I can share with you the local fixes I have for it.

Thanks for your time
Laura

**Comment by chixu** [comment]
same issue found here

**Comment by matthewww** [comment]
> I would like to know whether you think I am wrong in my thinking above, and we must always be very careful not to call stop from the onComplete callback.
I would say so for now Laura. 

In our own experiments, we saw that this is fixed by using unique IDs for the tweens within the library (as in the hashset branch). Perhaps this will be merged to master at some point. Rather than doing this though we have just stopped using .stop in onComplete.

**Comment by mikebolt** [comment]
It looks like this is the same as #264.

**Comment by lmeikle** [comment]
Hey Mike

I'm not 100% sure it is.
I believe this issue is caused by:
-stopping a tween which then removes it by splicing it out of the _tweens array
-at the same time the update function is still running iterating over the _tweens array
-as we have changed the size of the _tweens array (by removing something), the loop in the update function misses an element

I'm happy to wait and see if the hastset branch get released at some point, thanks matthewww for your comment.

I've fixed the issue in our local version by just setting the _tween element to null rather than splicing it out, and then the _update function null checks and splices at that point.

Thanks all
Laura
